### PR TITLE
[mp] add google oauth2 scopes to bigquery client

### PR DESCRIPTION
**Pull request by mattppal**
# Summary

Specify oauth2 scopes in Google BigQuery client to enable pulling from tables imported via Google Drive.

- [Scopes reference](https://developers.google.com/identity/protocols/oauth2/scopes)
- [dbt article](https://docs.getdbt.com/faqs/troubleshooting/access-gdrive-credential) with solution
- [Similar solution](https://stackoverflow.com/questions/53320537/creating-google-api-credentials-from-service-account-with-scope-and-delegated-a)

This closes https://github.com/mage-ai/mage-ai/issues/3022

image

image

# Tests
Locally with Mage Client.

### CODE REVIEW

File: mage_ai/io/bigquery.py [hunk 1]
```diff
@@ -37,36 +39,49 @@ def __init__(self, **kwargs) -> None:
...
-
-        credentials = kwargs.get('credentials')
+        if kwargs.get("verbose") is not None:
+            kwargs.pop("verbose")
+        super().__init__(verbose=kwargs.get("verbose", True))
+
+        CLIENT_SCOPES = [
+            "https://www.googleapis.com/auth/userinfo.email",
+            "https://www.googleapis.com/auth/cloud-platform",
+            "https://www.googleapis.com/auth/sqlservice.login",
```
Comment (wangxiaoyou1993) [inline mage_ai/io/bigquery.py]: Is this scope needed?

File: mage_ai/io/bigquery.py [hunk 2]
```diff
@@ -41,16 +43,28 @@ def __init__(self, **kwargs) -> None:
             kwargs.pop('verbose')
         super().__init__(verbose=kwargs.get('verbose', True))
 
+        CLIENT_SCOPES = [
+            'https://www.googleapis.com/auth/userinfo.email',
+            'https://www.googleapis.com/auth/cloud-platform',
+            'https://www.googleapis.com/auth/sqlservice.login',
+            'https://www.googleapis.com/auth/drive'
+
```
Comment (wangxiaoyou1993) [inline mage_ai/io/bigquery.py:51]: nit: remove empty line

File: mage_ai/io/bigquery.py (hunk 1 above)
Comment (mattppal) [reply mage_ai/io/bigquery.py]: https://cloud.google.com/bigquery/docs/external-data-drive#python

Looks like we might only need drive & bq. I borrowed these from dbt. I'll test

Comment (mattppal) [reply mage_ai/io/bigquery.py]: Confirmed this works with the following scopes, which match googles docs

```python
CLIENT_SCOPES = [
            "https://www.googleapis.com/auth/drive",
            "https://www.googleapis.com/auth/bigquery",
        ]
```

Comment (mattppal) [reply mage_ai/io/bigquery.py]: I would say I'm marginally concerned this could break something if a google drive api isn't enabled

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**